File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> mpp = new HashMap<>();
        mpp.put(0,1);
        int preSum=0, cnt=0;
        for(int i=0;i<nums.length;i++){
            preSum+=nums[i];
            cnt += mpp.getOrDefault(preSum-k, 0);
            mpp.put(preSum, mpp.getOrDefault(preSum,0)+1);
        }
        return cnt;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        h={0:1}
        preSum=0
        ans=0
        for i in nums:
            preSum+=i
            ans+=h.get(preSum-k,0)
            if preSum in h:
                h[preSum]+=1
            else:
                h[preSum]=1

        return ans